rumqttd/src/server/tls.rs

```diff
@@ -189,8 +191,16 @@ impl TLSAcceptor {
             // Get private key
             let key_file = File::open(key_path);
             let key_file = key_file.map_err(|_| Error::ServerKeyNotFound(key_path.clone()))?;
-            let keys = rustls_pemfile::rsa_private_keys(&mut BufReader::new(key_file));
-            let keys = keys.map_err(|_| Error::InvalidServerKey(key_path.clone()))?;
+            let mut key_file_buf = BufReader::new(key_file);
+            let mut keys = Vec::<Vec<u8>>::new();
+            for item in iter::from_fn(|| rustls_pemfile::read_one(&mut key_file_buf).transpose()) {
+                match item.unwrap() {
+                    rustls_pemfile::Item::RSAKey(key) => keys.push(key),
+                    rustls_pemfile::Item::PKCS8Key(key) => keys.push(key),
+                    rustls_pemfile::Item::ECKey(key) => keys.push(key),
+                    _ => return Err(Error::InvalidServerKey(key_path.clone())),
+                }
+            }
```

Comment:
- we only need one key, so we can avoid creating a `Vec<_>` and stop the loop once we get or first key!
- call to `item.unwrap()` should be avoided, we can return key not found or some other error.
- If read item is a cert, we still want to continue the loop, instead of returning error. This is how `rsa_private_key()` or other fns work, you can see their source in docs.
- That `transpose()` call can be avoided. we can use `loop{}` and `break` to read_one instead of creating iter.
- You can also refer to PR mentioned in issue related to this to get an clear idea!